### PR TITLE
[15.0][FW] queue_job: port queue dependencies from 14.0

### PR DESCRIPTION
**Pull request by guewen**
Port of https://github.com/OCA/queue/pull/403 from 14.0 to 15.0.

Can be tested on Runboat.

Open the runboat url with `/queue_job/create_test_job?size=30`.

image

### CODE REVIEW

File: queue_job_cron_jobrunner/models/queue_job.py [hunk 1]
```diff
@@ -105,6 +105,10 @@ def _process(self, commit=False):
             self.env["base"].flush()
             self.env.cr.commit()  # pylint: disable=invalid-commit
 
+        _logger.debug("%s enqueue depends started", job)
+        job.enqueue_waiting()
+        _logger.debug("%s enqueue depends done", job)
```
Comment (guewen) [inline queue_job_cron_jobrunner/models/queue_job.py:110]: I added this to enqueue dependencies cc @ivantodorovich

**Review by ivantodorovich — APPROVED** [review]
Thanks ❤️ !

**Review by leemannd — APPROVED** [review]
This is nice!

**Comment by jeremi** [comment]
Any idea on when this will be merged? This is a great improvement!

**Review by lmignon — APPROVED** [review]

**Comment by pedrobaeza** [comment]
A new copier update is needed due to https://github.com/OCA/oca-addons-repo-template/pull/170

**Pull request merged**